### PR TITLE
Issue-3: Add initial support for Tuple

### LINKED ISSUE #3: Add initial support for Tuple
Must include the following:

* A `Tuple` type.
* The following static constructors:
    * `of :: a → b → (a, b)`
* The following static methods:
    * `first :: (a, b) → a`: Extracts the first component of the pair.
    * `second :: (a, b) → b`: Extracts the second component of the pair.
    * `swap :: (a, b) → (b, a)`: Swaps the components of the pair.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #3
SemVer:Minor
enhancement

### CODE REVIEW

File: docs/data/Tuple/instance/toJSON.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Returns
+
+* `(String)`: The `instance` as a [`JSON`][JSON] formatted `String`.
+
+```javascript
+const tuple = Tuple.of("a", 1);
+
+tuple.toJSON();
+// => '{"tuple":[1, "a"]}'
```
Comment (doomchild) [inline docs/data/Tuple/instance/toJSON.md:13]: Why this format instead of `{"first": 1, "second": "a"}`?

Comment (jlmorgan) [reply docs/data/Tuple/instance/toJSON.md:13]: I wanted a means to declare the type of the thing as well as the thing itself, but not focus too much on the names of the underlying properties. I had considered what you had there as well as:

```javascript
{
  "couple": [1, "a"]
}
{
  "tuple": {
    "size": 2,
    "values": [1, "a"]
  }
}
```

And some others I cannot recall. The key thing with `Tuple` is that it is an _ordered_ sequence. This is why I chose an `Array` for the JSONified version, but I don't like the mixed typing. I am open to suggestions.

Comment (doomchild) [reply docs/data/Tuple/instance/toJSON.md:13]: If we have no plans to support anything beyond 2 items, I think the `size` field is redundant.  I don't have a huge emotional stake in this, but the format you documented just seemed different to what I had expected.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Pull request merged**